spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/V1LayerConverter.scala

```diff
@@ -0,0 +1,179 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.caffe
+
+import scala.collection.JavaConverters._
+import caffe.Caffe
+import caffe.Caffe.EltwiseParameter.EltwiseOp
+import caffe.Caffe.V1LayerParameter.LayerType
+import caffe.Caffe._
+import com.google.protobuf.GeneratedMessage
+import com.intel.analytics.bigdl.nn.Graph.ModuleNode
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{Node, Table}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+
+class V1LayerConverter[T: ClassTag](implicit ev: TensorNumeric[T]) extends Converter[T] {
+
+  override protected def fromCaffeConvolution(layer : GeneratedMessage) : Seq[ModuleNode[T]] = {
+    val param = getConvolutionParam(layer).get
+    val weightBlob = getBlob(layer, 0).get
+    val group = if (param.getGroup == 0)  1 else param.getGroup
+    val nInputPlane = weightBlob.getChannels * group
+    val nOutPlane = weightBlob.getNum
+    var kw = param.getKernelW
+    var kh = param.getKernelH
+    var dw = param.getStrideW
+    var dh = param.getStrideH
+    if (kw ==0 || kh == 0) {
+      kw = param.getKernelSize(0)
+      kh = kw
+    }
+    if (dw == 0 || dh == 0) {
+      if (param.getStrideList.size() != 0) {
+        dw = param.getStride(0)
+        dh = dw
+      } else {
+        // use default values if not found
+        dw = 1
+        dh = 1
+      }
+    }
+    var pw = param.getPadW
+    var ph = param.getPadH
+    if (pw == 0 || ph == 0) {
+      if (param.getPadList.size() != 0) {
+        pw = param.getPad(0)
+        ph = pw
+      }
+    }
+    Seq(new SpatialConvolution[T](nInputPlane, nOutPlane, kw, kh, dw, dh, pw, ph, group)
+      .setName(getLayerName(layer)).apply())
+  }
+
+  override protected def fromCaffeInnerProduct(layer : GeneratedMessage) : Seq[ModuleNode[T]] = {
+    val param = getInnerProductParam(layer).get
+    val withBias = param.getBiasTerm
+    val layerName = getLayerName(layer)
+    val weightBlob = getBlob(layer.asInstanceOf[V1LayerParameter], 0).get
+    var nInputPlane = weightBlob.getWidth
+    val nOutputPlane = param.getNumOutput
+    val linear = Linear[T](nInputPlane, nOutputPlane, withBias = withBias).setName(layerName)
+    val node = linear.apply()
+    if(nInputPlane != nOutputPlane) {
```

Comment:
this is to reshape the size, refer to https://github.com/szagoruyko/loadcaffe/blob/master/loadcaffe.cpp and also from pre-defined AlexNet module